### PR TITLE
Input events - Enable interacting with channel

### PR DESCRIPTION
**Pull request by cinderblock**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -160,9 +167,24 @@ class SSH {
             if (options.onStderr) options.onStderr(chunk)
             output.stderr.push(chunk)
           })
-          if (options.stdin) {
+          if (typeof options.stdin == 'string' && options.stdin) {
             stream.write(options.stdin)
             stream.end()
+          } else if (options.stdin instanceof EventEmitter) {
+            options.stdin.on('data', function(chunk) {
```
Comment (steelbrain) [inline src/index.js]: Instead of all of this, how about just `options.stdin.pipe(stream)`?

Comment (cinderblock) [reply src/index.js]: I was wondering how it would make sense to enable the stream api. This makes sense.

**Comment by cinderblock** [comment]
@steelbrain I was looking into using a stream based API instead. Unfortunately we can't simply use `options.stdin.pipe(stream)` as you suggested because I'm assuming `options.stdin` is simply an `EventEmitter` and not a `Stream.Readable`. It seems to take quite a bit more overhead/boilerplate (for the consumer of this API) to use a readable stream instead of an event emitter. If you don't have a stream already, you'll need to extend `Stream.Readable` and implement an emitter than otherwise is quite simple.

I do however think that enabling a stream based API would be valuable. This could be made a bi-directional stream and handle stdout as well.

**Comment by steelbrain** [comment]
Thank you VERY much @cinderblock for this PR. I appreciate your effort a lot.

This PR is solving a very specific case, while it's nice to be able to solve specific cases, they pile up after a while and maintenance becomes difficult. Therefore I've added a compromise in https://github.com/steelbrain/node-ssh/pull/297. It adds a `onChannel` callback, where you can implement this exact functionality but inside the consumer instead.

Closing, but feel free to continue discussing

**Pull request closed without merging**